mypy/checker.py

```diff
@@ -2275,6 +2276,26 @@ def visit_decorator(self, e: Decorator) -> None:
         if e.func.is_property:
             self.check_incompatible_property_override(e)
 
+    def check_for_untyped_decorator(self,
+                                    func: FuncDef,
+                                    dec_type: Type,
+                                    dec_expr: Expression) -> None:
+        def is_typed_callable(c: Optional[Type]) -> bool:
+            if not c or not isinstance(c, CallableType):
+                return False
+            return all(not isinstance(t, AnyType) or not t.implicit
```

Comment:
Nit: I think this would be much clearer as `not any(isinstance(t, AnyType) and t.implicit)`.